src/test/run-pass/issue-7222.rs

```diff
@@ -10,11 +10,20 @@
 
 // pretty-expanded FIXME #23616
 
+#![allow(illegal_floating_point_constant_pattern)]
+
 pub fn main() {
-    const FOO: f64 = 10.0;
+    const FOO: u8 = 10;
+
+    match 0 {
+        0 ... FOO => (),
+        _ => ()
+    }
+
+    const FOOFL: f64 = 10.0;
 
     match 0.0 {
-        0.0 ... FOO => (),
+        0.0 ... FOOFL => (),
```

Comment:
This, i feel, should keep working just fine. It is just a range and testing that floats fall into some range is encouraged even.